backend/infrahub/core/migrations/graph/m020_duplicate_edges.py

```diff
@@ -0,0 +1,174 @@
+// -------------------
+// get/create the one edge to keep
+// -------------------
+CREATE (a)-[fresh_e:HAS_VALUE {branch: branch, branch_level: branch_level, status: status, from: from}]->(the_one_av)
+SET fresh_e.to = to
+WITH a, branch, status, from, to, attr_val, attr_default, %(id_func)s(fresh_e) AS e_id_to_keep
+// -------------------
+// get the identical edges for a given set of Attribute node, edge properties, AttributeValue.value
+// -------------------
+CALL {
+    // -------------------
+    // delete the duplicate edges a given set of Attribute node, edge properties, AttributeValue.value
+    // -------------------
+    WITH a, branch, status, from, to, attr_val, attr_default, e_id_to_keep
+    MATCH (a)-[e:HAS_VALUE]->(av:AttributeValue {value: attr_val, is_default: attr_default})
+    WHERE %(id_func)s(e) <> e_id_to_keep
+    AND e.branch = branch AND e.status = status AND e.from = from
+    AND (e.to = to OR (e.to IS NULL AND to IS NULL))
+    DELETE e
+}
+// -------------------
+// delete any orphaned AttributeValue nodes
+// -------------------
+WITH NULL AS nothing
+LIMIT 1
+MATCH (orphaned_av:AttributeValue)
+WHERE NOT exists((orphaned_av)-[]-())
+DELETE orphaned_av
+        """ % {"id_func": db.get_id_function_name()}
+        self.add_to_query(query)
+
+
+class DeleteDuplicateIsVisibleEdgesQuery(Query):
+    name = "delete_duplicate_is_visible_edges"
+    type = QueryType.WRITE
+    insert_return = False
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:
+        query = """
+// -------------------
+// find Attribute nodes with multiple identical edges to Boolean nodes
+// -------------------
+MATCH (a:Attribute)-[e:IS_VISIBLE]->(b)
+WITH a, e.branch AS branch, e.branch_level AS branch_level, e.status AS status, e.from AS from, e.to AS to, b, COUNT(*) AS num_duplicate_edges
+WHERE num_duplicate_edges > 1
+// -------------------
+// get the identical edges for a given set of Attribute node, edge properties, Boolean
+// -------------------
+WITH DISTINCT a, branch, branch_level, status, from, to, b
+CREATE (a)-[fresh_e:IS_VISIBLE {branch: branch, branch_level: branch_level, status: status, from: from}]->(b)
+SET fresh_e.to = to
+WITH a, branch, status, from, to, b, %(id_func)s(fresh_e) AS e_id_to_keep
+CALL {
+    WITH a, branch, status, from, to, b, e_id_to_keep
+    MATCH (a)-[e:IS_VISIBLE]->(b)
+    WHERE %(id_func)s(e) <> e_id_to_keep
+    AND e.branch = branch AND e.status = status AND e.from = from
+    AND (e.to = to OR (e.to IS NULL AND to IS NULL))
+    DELETE e
+}
+        """ % {"id_func": db.get_id_function_name()}
+        self.add_to_query(query)
+
+
+class DeleteDuplicateIsProtectedEdgesQuery(Query):
+    name = "delete_duplicate_is_protected_edges"
+    type = QueryType.WRITE
+    insert_return = False
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:
+        query = """
+// -------------------
```

Comment:
This query looks almost identical to the previous one, could we consolidate them and pass the name of the edge as an argument ?